components/style/style_resolver.rs

```diff
@@ -298,6 +298,28 @@ where
         })
     }
 
+    /// Cascade a set of rules for pseudo element, using the default parent for inheritance.
+    pub(crate) fn cascade_style_and_visited_for_pseudo_with_default_parents(
+        &mut self,
+        inputs: CascadeInputs,
+        pseudo: &PseudoElement,
+        primary_style: &PrimaryStyle,
+    ) -> ResolvedStyle {
+        with_default_parent_styles(self.element, |_, layout_parent_style| {
+            let layout_parent_style_for_pseudo = if primary_style.style().is_display_contents() {
+                layout_parent_style
+            } else {
+                Some(primary_style.style())
+            };
+
+            self.cascade_style_and_visited(
+                inputs,
+                Some(primary_style.style()),
+                layout_parent_style_for_pseudo,
+                Some(pseudo),
+            )
+        })
+    }
```

Comment:
nit: newline missing.

Reply:
Fixed. It's curious that tidy didn't correct this.